bittensor/_dataset/dataset_impl.py

```diff
@@ -1,538 +1,494 @@
 
-        Returns:
-            text: str: 
-                Contents of the text data.
-        """
-        self.IPFS_fails = 0
-        data_corpus = []
-        try:
-            # --- Get directories from a random dataset_hash
-            directories = list(self.get_hashes_from_dataset())
-
-            # --- Generate a random order of the directories
-            random.shuffle(directories)
-
-            # --- Pick random directories and get their text contents.
-            if directories:
-                total_dataset_size = 0
-                total_dataset_len = 0
-                i = 0
-
-                # --- Dont stop until the corpus size and the minimum data_length was reached.
-                n_workers = cpu_count() if self.num_workers == 0 else self.num_workers
-                with concurrent.futures.ThreadPoolExecutor(max_workers=n_workers) as executor:
-                    future_map = {}
-                    for idx, call_arg in enumerate(directories[:self.max_directories]):
-                        future = executor.submit(self.get_text, call_arg)
-                        future_map[future] = call_arg
-
-                    for i, future in enumerate(concurrent.futures.as_completed(future_map)):
-                        text = future.result()
-
-                        if text is not None:
-                            text_list = text.split()
-                            data_corpus.extend(text_list)
-                            total_dataset_len += len(text_list)
-
-                        if (total_dataset_len > min_data_len) or self.IPFS_fails > self.IPFS_fails_max:
-                            break
+        Returns 
+            output (dict, list) 
+                The output of the loaded JSON.
 
-            else:
-                logger.error("It appears the directory is empty... Restart your miner to try again.")
+        '''
+        output =  self.async_run(job=self.async_load_json(*args,**kwargs), loop=loop)
+
+        return output
 
-        except Exception as e:
-            logger.error("Ran into exception when trying to retrieve dataset from IPFS: {}".format(e))
+    async def async_build_single_dataset(self, dataset:str = None, num_folders = None, num_samples:int = 10000, save:bool=False, load:bool=True, loop: 'asyncio.loop' =None) -> List[dict] :
+        """ Building a single dataset by fetching its text file metas ({Hash:str, Name:str, Size: int})
+        Args:
+            dataset (List[str]):
+                The name of the dataset
+            num_folders (bool):
+                The numbe of folders in the dataset.
+            num_samples (int):
+                The number of samples the user want so get from the dataset
+            load (bool):
+                Load the dataset hashes locally
+            save (bool):
+                Save the dataset hahses locally.
+            loop (asyncio.Loop):
+                Asyncio loop 
+
+        Returns: None
```

Comment:
This docstring is also a lie.

Personally I'm in favour dropping type hints in docstrings (@unconst, thoughts?). It's redundant and not in a good way. They're easy to miss when refactoring, and tbh the type hints in the code should just tell the story themselves. This also facilitates static type checking w/ `mypy` or similar, which I'm strongly in favour of adding to our dev workflow.

Reply:
fixed the docstring